plugin/ipam/driver.go

```diff
@@ -81,11 +81,21 @@ func (i *ipam) RequestAddress(poolID string, address net.IP, options map[string]
 	if _, iprange, err = net.ParseCIDR(parts[2]); err != nil {
 		return
 	}
-	// We are lying slightly to IPAM here: the range is not a subnet
-	if ip, err = i.weave.AllocateIPInSubnet("_", iprange); err != nil {
-		return
+	// See if we can give preferred address
+	if address != nil {
+		if err := i.weave.ClaimIP("_", address); err == nil {
+			ip = &net.IPNet{IP: address, Mask: subnet.Mask}
+		} else {
+			Log.Debugln("ClaimIP returned", err, "; allocating new address")
```

Comment:
Is this the right thing to do, or should we fail the `RequestAddress` operation?